src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/JintLuceneDocumentConverter.cs

```diff
@@ -144,36 +186,87 @@ protected override int GetFields<T>(T instance, LazyStringValue key, LazyStringV
             }
 
             return newFields;
+
+            static bool IsObject(JsValue value)
+            {
+                return value.IsObject() && value.IsArray() == false;
+            }
+
+            static object CreateValueForIndexing(object value, float? boost)
+            {
+                if (boost.HasValue == false)
+                    return value;
+
+                return new BoostedValue
+                {
+                    Boost = boost.Value,
+                    Value = value
+                };
+            }
+
+            static void BoostDocument(T instance, int numberOfCreatedFields, float? boost)
```

Comment:
Why do we care about the number of fields?

Reply:
This is copied from the anonymous one. We are not boosting 'auxiliary' fields like `id()` or `reduceKey()`